Git-commit: 9320e087f2b64257f34106eecbfe5a43be5199b0
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Tue, 10 May 2022 15:20:42 +0200
Subject: [PATCH] random: use proper jiffies comparison macro
References: bsn#19
Patch-mainline: v5.10.119

commit 8a5b8a4a4ceb353b4dd5bafd09e2b15751bcdb51 upstream.

This expands to exactly the same code that it replaces, but makes things
consistent by using the same macro for jiffy comparisons throughout.

Signed-off-by: Jason A. Donenfeld <Jason@zx2c4.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/char/random.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/char/random.c b/drivers/char/random.c
index f1c60f486196..40ad91e6e88e 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -326,7 +326,7 @@ static bool crng_has_old_seed(void)
 			interval = max_t(unsigned int, CRNG_RESEED_START_INTERVAL,
 					 (unsigned int)uptime / 2 * HZ);
 	}
-	return time_after(jiffies, READ_ONCE(base_crng.birth) + interval);
+	return time_is_before_jiffies(READ_ONCE(base_crng.birth) + interval);
 }
 
 /*
-- 
2.35.1

